src/bootstrap/src/core/build_steps/check.rs

```diff
@@ -82,22 +82,16 @@ impl Step for Std {
             format_args!("library artifacts{}", crate_description(&self.crates)),
             target,
         );
-        run_cargo(
-            builder,
-            cargo,
-            builder.config.free_args.clone(),
-            &libstd_stamp(builder, compiler, target),
-            vec![],
-            true,
-            false,
-        );
+
+        let stamp = build_stamp::libstd_stamp(builder, compiler, target).with_prefix("check");
+        run_cargo(builder, cargo, builder.config.free_args.clone(), &stamp, vec![], true, false);
```

Comment:
Changes in 7b72325607226f3ac9349776de0cea12c68eb0ff for check.rs will somehow mangle prefix names?
Before: `.libstd-check-stamp`
After: `.check-libstd-stamp`

Reply:
I did this intentionally to keep it simple by avoiding the need to duplicate similar functions. I assume this rename shouldn't cause any issues. Maybe, at worst, it might require running `x clean` after pulling these changes to remove old build artifacts.

Reply:
Yeah that makes sense